server/bleep/src/webserver/answer.rs

```diff
@@ -915,15 +924,12 @@ impl Conversation {
             s
         };
 
-        let query_history = self.query_history().join("\n");
-        let query = self
-            .last_exchange()
-            .query()
-            .context("exchange did not have a user query")?;
-
-        let prompt = prompts::final_explanation_prompt(&context, query, &query_history);
-
-        let messages = [llm_gateway::api::Message::system(&prompt)];
+        let query_history = self.query_history().collect::<Vec<_>>();
+        let system_message = prompts::final_explanation_prompt(&context);
+        let messages = Some(llm_gateway::api::Message::system(&system_message))
```

Comment:
Is there a concern that we'll run out of prompt space if we put _all_ the query, answer pairs in the prompt? We could grab the last `n` (5?), or populate until we run out of prompt space.

Reply:
In this case, we're using the conclusions only, which tend to be much shorter. Message contents should be identical compared to `main`.

Reply:
Would we be putting whole explanations in? These could be much longer than our current conclusions. We want to support cases where the agent explains how a function works, and the user responds `can you explain that in more detail`

Reply:
To be clear, this PR does not add whole explanations to the history, for the token concerns you noted. We can do that, but this specifically is a refactoring change designed to help #688. We get better delegation between user and assistant messages.

Did you want to include whole explanations in this PR?

Reply:
Understood. Nah, let's add explanations in #688